### PR TITLE
use github matrix for CI

### PR DESCRIPTION
**Pull request by bplommer**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,16 +2,18 @@ name: ci
 on:
   push:
     branches: ['*']
-  pull_request:
```
Comment (vlovgr) [inline .github/workflows/ci.yml:5]: Deleting this means that we no longer build pull requests (coming from a fork) raised against the repository.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -2,16 +2,18 @@ name: ci
...
     - uses: actions/checkout@v2
       with:
         fetch-depth: 0
     - uses: olafurpg/setup-scala@v10
     - uses: coursier/cache-action@v5
-    - run: sbt validate
-    - uses: codecov/codecov-action@v1
+    - run: sbt ++${{ matrix.scala }} validate
+    - if: matrix.scala == '2.13.3'
+      run: sbt "++${{ matrix.scala }} docs/run"
```
Comment (vlovgr) [inline .github/workflows/ci.yml:19]: I'm not a fan of having scala versions here, and essentially overriding whatever is defined in `build.sbt`.

File: build.sbt [hunk 3]
```diff
@@ -310,15 +310,12 @@ def addCommandsAlias(name: String, values: List[String]) =
 addCommandsAlias(
   "validate",
   List(
```
Comment (vlovgr) [inline build.sbt:312]: Sadly, these changes mean you can no longer just run `sbt validate` (you have to run `sbt +validate`) locally to check everything passes.

**Comment by vlovgr** [comment]
@bplommer I think it would be good to have parallel tests for time-saving reasons. But would like to investigate using something like [sbt-github-actions](https://github.com/djspiewak/sbt-github-actions) to avoid the issues in my comments above. In the meantime I'm reverting this with #471.

**Comment by bplommer** [comment]
> @bplommer I think it would be good to have parallel tests for time-saving reasons. But would like to investigate using something like [sbt-github-actions](https://github.com/djspiewak/sbt-github-actions) to avoid the issues in my comments above. In the meantime I'm reverting this with #471.

Sounds reasonable - I'll look into using sbt-github-actions.

Sorry, should have left this open for review.

**Pull request merged**